core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java

```diff
@@ -65,9 +67,9 @@
 
   private ScannerImpl.Reporter reporter;
 
-  private static ThreadPoolExecutor readaheadPool = ThreadPools.createThreadPool(0,
```

Comment:
I did not realize this pool was static and shared among all scanners. I wonder if changing this to per-scanner could improve performance for the case of a client process w/ many concurrent scanners.

Reply:
> If you notice here the max threads is Integer.MAX_VALUE. Now that this is not static anymore, what should the max value be?

oh I completely missed the int.max_val.  I saw a 3 and thought that was the max. So maybe there will not be a performance diff.  The new max should probably be lower. The scan code may never submit more than 1 or 2 task to the pool.  Not sure, would need to look at that code.

Reply:
Do you think we should reuse the value of `BATCH_SCANNER_NUM_QUERY_THREADS` for this, create a new property `SCANNER_NUM_QUERY_THREADS`, or something else?

Reply:
> Do you think we should reuse the value of BATCH_SCANNER_NUM_QUERY_THREADS for this, create a new property SCANNER_NUM_QUERY_THREADS, or something else?

Looking at [this code](https://github.com/apache/accumulo/blob/bf4d8e4ad5c06e2881a8235ffd0bb293c54289e4/core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java#L143-L146) it seems like the read ahead thread pool on a per scanner basis would only ever use a max one thread.  With that in mind I don't think we need to make the pool configurable, just create a pool with 0 to 1 threads.

Reply:
The user has control over wether read ahead will happen w/ [this scanner API](https://github.com/apache/accumulo/blob/bf4d8e4ad5c06e2881a8235ffd0bb293c54289e4/core/src/main/java/org/apache/accumulo/core/client/Scanner.java#L93).  If that is set really high, then read ahead would never happen.  So maybe it nice if the thread pool can go to zero threads instead of always having 1 thread.

Reply:
So maybe nothing needs to be done here. The core pool size is already being set to [zero](https://github.com/apache/accumulo/pull/2346/files#diff-4c63ba161b875d96d2612de9bfc7cf93649762e2034d0700ec4fbfe6de0d6ef0R144).